src/main/java/seedu/address/storage/JsonCourseParser.java

```diff
@@ -0,0 +1,68 @@
+package seedu.address.storage;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import seedu.address.model.course.Course;
+
+/**
+ * JsonCourseParser is a utility class for parsing a JSON file containing course data
+ * and extracting module codes into a list of strings.
+ */
+public class JsonCourseParser {
+
+    public static final Path COURSE_PATH = Paths.get("data", "JSONModuleList.json");
+
+    /**
+     * Parses a JSON file and extracts module codes into a list of strings.
+     *
+     * @param jsonFilePath The path to the JSON file to be parsed.
+     * @return A list of module codes extracted from the JSON file.
+     * @throws IOException If an I/O error occurs or if the JSON file does not exist or has invalid format.
+     */
+    public static List<String> parseJsonFile(Path jsonFilePath) throws IOException {
```

Comment:
Maybe can change the List<String> to Set<String> instead? hash set set lookup is O(1) and may give better performance esp. since we have so many courses.

Reply:
Hello Joel! I have changed it to a use a HashSet instead, increasing it's performance.